pinot-controller/src/main/java/org/apache/pinot/controller/api/upload/ZKOperator.java

```diff
@@ -276,6 +336,144 @@ private void processExistingSegment(String tableNameWithType, SegmentMetadata se
     }
   }
 
+  // process a batch of existing segments
```

Comment:
Are we not able to leverage the existing method processExistingSegment for a single segment, to avoid duplication of code ? Basically reuse this method as iterate over list of segments. 

Same for processNewSegments and other methods in ZkOperator.

Reply:
Originally, I had modified the existing methods to deal with a list of segments as opposed to a single segment. Later decided to use separate methods for single segment upload vs batch upload as recommended by Jackie. His original comment as below.
```
Consider using different method to handle single segment upload vs batch segment upload. We may extract common part as needed, but modifying the existing flow can be risky. It can cause backward incompatible if any error handling behavior is changed
```